components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/meta/MongoDBMetaExtension.java

```diff
@@ -0,0 +1,104 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.mongodb.meta;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.client.model.Filters;
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.extension.metadata.AbstractMetaDataExtension;
+import org.apache.camel.component.extension.metadata.MetaDataBuilder;
+import org.apache.camel.component.mongodb.MongoDbComponent;
+import org.apache.camel.component.mongodb.conf.MongoConfiguration;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.camel.util.CastUtils.cast;
+
+public class MongoDBMetaExtension extends AbstractMetaDataExtension {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongoDBMetaExtension.class);
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    public MongoDBMetaExtension() {
+        this(null);
+    }
+
+    public MongoDBMetaExtension(CamelContext context) {
+        super(context);
+    }
+
+    @Override
+    public Optional<MetaData> meta(Map<String, Object> parameters) {
+        Map<String, String> textParameters = cast(parameters);
+        LOGGER.debug("Fetching mongodb meta information with params: {}", textParameters);
+
+        MongoConfiguration mongoConf = new MongoConfiguration(textParameters);
+        MongoClientURI connectionURI = new MongoClientURI(mongoConf.getMongoClientURI());
+
+        JsonNode collectionInfoRoot = null;
+        try (MongoClient mongoConnection = new MongoClient(connectionURI)) {
+            Document collectionInfo = mongoConnection.getDatabase(textParameters.get("database"))
+                    .listCollections()
+                    .filter(Filters.eq("name", textParameters.get("collection"))).first();
+            LOGGER.debug("Collection info: {}", collectionInfo);
+            if (collectionInfo == null) {
+                LOGGER.error("Cannot read information for collection {}.{}", textParameters.get("database"),
+                        textParameters.get("collection"));
+                return Optional.empty();
+            }
+            String collectionInfoJson = collectionInfo.toJson();
+            collectionInfoRoot = objectMapper.readTree(collectionInfoJson.replaceAll("bsonType", "type"));
+        } catch (IOException e) {
+            LOGGER.error("Error occurred while reading schema information", e);
+            return Optional.empty();
+        }
+
+        JsonNode schemaRoot = collectionInfoRoot.path("options").path("validator").path("$jsonSchema");
```

Comment:
Do you think if it makes sense to check for `collectionInfoRoot` if it is not `null` before invoking `.path` on it?

Reply:
It should not be necessary. The value is taken originally from `collectionInfo` and there is already a `null` checking on that.

Reply:
Aha you are right, misses that